tests/framework/validators/UrlValidatorTest.php

```diff
@@ -28,6 +28,14 @@ public function testValidateValue()
                                         .'&rls=org.mozilla:de:official&client=firefox-a&gws_rd=cr'));
         $this->assertFalse($val->validate('ftp://ftp.ruhr-uni-bochum.de/'));
         $this->assertFalse($val->validate('http://invalid,domain'));
+        $this->assertFalse($val->validate('http://example.com,'));
+        $this->assertFalse($val->validate('http://example.com*12'));
+        $this->assertTrue($val->validate('http://example.com/*12'));
+        $this->assertTrue($val->validate('http://example.com#test'));
+        $this->assertTrue($val->validate('http://example.com/?test'));
+        $this->assertTrue($val->validate('http://example.com?test'));
+        $this->assertTrue($val->validate('http://example.com/#test'));
+        $this->assertFalse($val->validate('http://example.com,?test'));
```

Comment:
These test are redundant because only the first char after the domain name has limitations